privacyidea/lib/eventhandler/scripthandler.py

```diff
@@ -42,23 +43,31 @@
 
 log = logging.getLogger(__name__)
 
+SCRIPT_BACKGROUND = "Run in background"
+SCRIPT_WAIT = "Wait for script to finish"
+
 
 class ScriptEventHandler(BaseEventHandler):
     """
     An Eventhandler needs to return a list of actions, which it can handle.
     It also returns a list of allowed action and conditions
-    It returns an identifier, which can be used in the eventhandlig definitions
+    It returns an identifier, which can be used in the eventhandling definitions
     """
 
     identifier = "Script"
     description = "This event handler can trigger external scripts."
 
-    try:
-        script_directory = get_from_config("PI_SCRIPT_HANDLER_DIRECTORY",
-                                           "/etc/privacyidea/scripts")
-    except RuntimeError:
-        # In case of the tests we are outside of the application context
-        script_directory = "tests/testdata/scripts"
+    def __init__(self, script_directory=None):
+        if not script_directory:
+            try:
+                self.script_directory = get_from_config("PI_SCRIPT_HANDLER_DIRECTORY",
+                                                        "/etc/privacyidea/scripts")
+            except RuntimeError:
```

Comment:
Please use `except RuntimeError as e:` since Python 3 doesn't allow the old syntax.

Reply:
Done in bea2309